RecoVertex/PrimaryVertexProducer_Alpaka/plugins/alpaka/PortableTrackSoAProducer.cc

```diff
@@ -0,0 +1,244 @@
+                });
+
+      int32_t nTrueTracks =
+          0;  // This will keep track of how many we actually copy to device, only those that pass filter
+      for (int32_t idx = 0; idx < tsize_; idx++) {
+        // Fill up the the Track SoA, weight doubles up as an isGood flag, as we compute it only for good tracks
+        double weight = convertTrack(tview[nTrueTracks],
+                                     sortedTracksPair[idx].second,
+                                     beamSpot,
+                                     fParams,
+                                     sortedTracksPair[idx].first,
+                                     nTrueTracks);
+        if (weight > 0) {
+          nTrueTracks += 1;
+          tview.nT() += 1;
+          tview.totweight() += weight;
+        }
+      }
+#ifdef DEBUG_RECOVERTEX_PRIMARYVERTEXPRODUCER_ALPAKA_PORTABLETRACKSOAPRODUCER
+      printf("[PortableTrackSoAProducer::produce()] From %i tracks, %i pass filters\n",
+             (int32_t)tracks->size(),
+             nTrueTracks);
+#endif
+      // Create device collections and copy into device
+      portablevertex::TrackDeviceCollection deviceTracks{tsize_, iEvent.queue()};
+
+      alpaka::memcpy(iEvent.queue(), deviceTracks.buffer(), hostTracks.buffer());
+
+      // And put into the event
+      iEvent.emplace(devicePutToken_, std::move(deviceTracks));
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::InputTag>("TrackLabel");
+      desc.add<edm::InputTag>("BeamSpotLabel");
+      edm::ParameterSetDescription psd0;
+      psd0.add<double>("maxNormalizedChi2", 10.0);
+      psd0.add<double>("minPt", 0.0);
+      psd0.add<std::string>("algorithm", "filter");
+      psd0.add<double>("maxEta", 2.4);
+      psd0.add<double>("maxD0Significance", 4.0);
+      psd0.add<double>("maxD0Error", 1.0);
+      psd0.add<double>("maxDzError", 1.0);
+      psd0.add<std::string>("trackQuality", "any");
+      psd0.add<int>("minPixelLayersWithHits", 2);
+      psd0.add<int>("minSiliconLayersWithHits", 5);
+      psd0.add<double>("vertexSize", 0.006);
+      psd0.add<double>("d0CutOff", 0.10);
+      desc.add<edm::ParameterSetDescription>("TkFilterParameters", psd0);
+      descriptions.addWithDefaultLabel(desc);
+    }
+
+  private:
+    edm::EDGetTokenT<reco::TrackCollection> trackToken_;
+    edm::EDGetTokenT<reco::BeamSpot> beamSpotToken_;
+    const edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> theTTBToken;
+    device::EDPutToken<portablevertex::TrackDeviceCollection> devicePutToken_;
+    edm::ParameterSet theConfig;
+    static double convertTrack(portablevertex::TrackHostCollection::View::element out,
+                               const reco::TransientTrack in,
+                               const reco::BeamSpot bs,
+                               filterParameters fParams,
```

Comment:
I find it a bit unusual to define `convertTrack` as a `static` function, and then pass the `fParams` data member as an argument. Technically it is ok, but the alternative of a normal member function would allow to remove `fParams` argument.